probe-rs/src/debug/mod.rs

```diff
@@ -738,53 +789,61 @@ impl DebugInfo {
-                    .create_function_scope_cache(core, function_die, &unit_info)
-                    .map_or_else(
-                        |error| {
-                            log::error!(
-                                "Could not resolve function variables. {}. Continuing...",
-                                error
-                            );
-                            None
-                        },
-                        Some,
-                    );
+                    // Next, resolve and cache the function variables.
+                    let local_variables = self
+                        .create_function_scope_cache(core, function_die, &unit_info)
+                        .map_or_else(
+                            |error| {
+                                log::error!(
+                                    "Could not resolve function variables. {}. Continuing...",
+                                    error
+                                );
+                                None
+                            },
+                            Some,
+                        );
 
-                frames.push(StackFrame {
-                    // MS DAP Specification requires the id to be unique accross all threads, so using  so using unique `Variable::variable_key` of the `stackframe_root_variable` as the id.
-                    id: get_sequential_key(),
-                    function_name,
-                    source_location: inlined_caller_source_location,
-                    registers: stack_frame_registers.clone(),
-                    pc: inlined_call_site.unwrap(),
-                    is_inlined: function_die.is_inline(),
-                    static_variables,
-                    local_variables,
-                });
+                    frames.push(StackFrame {
+                        // MS DAP Specification requires the id to be unique accross all threads, so using  so using unique `Variable::variable_key` of the `stackframe_root_variable` as the id.
+                        id: get_sequential_key(),
+                        function_name,
+                        source_location: inlined_caller_source_location,
+                        registers: stack_frame_registers.clone(),
+                        pc: inlined_call_site,
+                        is_inlined: function_die.is_inline(),
+                        static_variables,
+                        local_variables,
+                    });
+                } else {
+                    log::warn!(
+                        "UNWIND: Unknown call site for inlined function {}.",
+                        function_name
+                    );
+                }
             }
 
             // Handle last function, which contains no further inlined functions
+            #[allow(clippy::unwrap_used)]
             let last_function = functions.last().unwrap(); //UNWRAP: Checked at beginning of loop, functions must contain at least one value
```

Comment:
Please put comments like this on the line above for shorter lines. IT also makes future edits easier :)